src/Caching/Storages/FileStorage.php

```diff
@@ -143,7 +143,7 @@ public function lock($key)
 	{
 		$cacheFile = $this->getCacheFile($key);
 		if ($this->useDirs && !is_dir($dir = dirname($cacheFile))) {
-			@mkdir($dir); // @ - directory may already exist
+			@mkdir($dir, 0777, true); // @ - directory may already exist
```

Comment:
@dg Let's say only a `dir/` exists and 1st thread is creating `dir/aaa/bbb/` and 2nd `dir/aaa/ccc/`. When they start in the same time, 1st creates `dir/aaa/` and 2nd tries too. 1st is success and continues to create `dir/aaa/bbb/` but 2nd fails and ends, so `dir/aaa/ccc/` does not exist.